llama-index-packs/llama-index-packs-rag-fusion-query-pipeline/llama_index/packs/rag_fusion_query_pipeline/base.py

```diff
@@ -80,7 +80,9 @@ def __init__(
             splitter = SentenceSplitter(chunk_size=chunk_size, chunk_overlap=0)
             nodes = splitter.get_nodes_from_documents(documents)
 
-            service_context = ServiceContext.from_defaults(llm=self.llm)
+            service_context = ServiceContext.from_defaults(
```

Comment:
should we even be using service context? We can pass the embed model to the index, and the llm to the query engine

```
vector_index = VectorStoreIndex(nodes, embed_model=embed_model)
self.query_engines.append(vector_index.as_query_engine(llm=self.llm))
```

Reply:
Indeed, `service_context` is marked as deprecated.